public/app.js

```
@@ -2181,13 +2203,49 @@ function setChallengeProfileName(name) {
 if (challengeProfileInputEl) {
-  challengeProfileInputEl.value = getChallengeProfile().name;
+  // Prefill: keep any stored localStorage name IF it still passes
+  // the unified validator; otherwise drop in a regex-clean random
+  // default. The validator-on-load step matters because localStorage
+  // may carry over a legacy 25-64 char name or one with characters
+  // the old regex allowed but the new one doesn't (digits in older
+  // challenge sessions). Starting in an invalid state would 400 on
+  // submit until the user manually edits. Caught by Copilot on
+  // PR #180.
+  const storedChallengeName = getChallengeProfile().name;
+  const initialName = normalizeProfileName(storedChallengeName) || pickRandomName();
```

ESLint

[error] 'pickRandomName' is not defined.

(no-undef)